Px.Utils/Validation/SyntaxValidation/SyntaxValidationUtilityMethods.cs

```diff
@@ -0,0 +1,404 @@
+        }
+
+        /// <summary>
+        /// Finds the keyword separators in a string. Only indeces that are not enclosed within strings are returned.
+        /// </summary>
+        /// <param name="input">The input string to search</param>
+        /// <param name="syntaxConf">The syntax configuration for the PX file. The syntax configuration is represented by a <see cref="PxFileSyntaxConf"/> object.</param>
+        /// <returns>An array of integers representing the indeces of keyword separators which are not enclosed by string delimeters from the input string</returns>
+        public static int[] FindKeywordSeparatorIndeces(string input, PxFileSyntaxConf syntaxConf)
+        {
+            Dictionary<int, int> stringIndeces = GetEnclosingCharacterIndexes(input, syntaxConf.Symbols.Key.StringDelimeter);
+            List<int> separators = [];
+
+            // Find all indeces of syntaxConf.Symbols.KeywordSeparator in the input string that are not enclosed within strings using FindSymbolIndex method
+            int separatorIndex = -1;
+            do
+            {
+                separatorIndex = FindSymbolIndex(input, syntaxConf.Symbols.KeywordSeparator, stringIndeces, separatorIndex + 1);
+                if (separatorIndex != -1)
+                {
+                    separators.Add(separatorIndex);
+                }
+            }
+            while (separatorIndex != -1);
+
+            return [..separators];
+        }
+
+        /// <summary>
+        /// Finds the illegal characters in a string using a regular expression pattern.
+        /// </summary>
+        /// <param name="input">The input string to search</param>
+        /// <param name="illegalCharacters">An array of characters that are considered illegal</param>
+        /// <param name="foundIllegalCharacters">An array of characters that were found in the input string</param>
+        /// <returns>Returns a boolean which is true if the search was successful. If the search times out, false is returned.</returns>
+        public static bool FindIllegalCharactersInString(string input, char[] illegalCharacters, out char[] foundIllegalCharacters, int customTimeout = DEFAULT_TIMEOUT)
+        {
+            string processedCharacters = ProcessCharsForRegexPattern(illegalCharacters);
+            string pattern = $"[{processedCharacters}]";
+
+            TimeSpan timeout = TimeSpan.FromSeconds(customTimeout);
+            try
+            {
+                MatchCollection matches = Regex.Matches(input, pattern, RegexOptions.Singleline, timeout);
+                foundIllegalCharacters = matches.Cast<Match>().Select(m => m.Value[0]).ToArray();
+                return true;
+            }
+            catch (RegexMatchTimeoutException)
+            {
+                foundIllegalCharacters = [];
+                return false;
+            }
+        }
+        private static int GetNextLineChangeIndex(string input, PxFileSyntaxConf syntaxConf, int startIndex = 0)
```

Comment:
What is the input this function expects?

Reply:
Removed this